pkg/azure/client.go

```diff
@@ -120,22 +132,80 @@ func (c *azureDNS) Delete(ctx context.Context, record *phonebook.DNSRecord) erro
 
 // Convert a DNSRecord to an Azure DNS record set
 func (c *azureDNS) resourceRecordSet(record *phonebook.DNSRecord) armdns.RecordSet {
+	ttl := defaultTTL
+	if record.Spec.TTL != nil {
+		ttl = *record.Spec.TTL
+	}
+
 	params := armdns.RecordSet{
 		Properties: &armdns.RecordSetProperties{
-			TTL: to.Ptr(int64(60)),
+			TTL: to.Ptr(int64(ttl)),
 		},
 	}
 
-	// Create specific record types based on the DNS type (e.g., A, CNAME)
+	// Create specific record types based on the DNS type
 	switch armdns.RecordType(record.Spec.RecordType) {
 	case armdns.RecordTypeA:
-		params.Properties.ARecords = []*armdns.ARecord{
-			{IPv4Address: to.Ptr(record.Spec.Targets[0])},
+		aRecords := make([]*armdns.ARecord, len(record.Spec.Targets))
+		for i, target := range record.Spec.Targets {
+			aRecords[i] = &armdns.ARecord{IPv4Address: to.Ptr(target)}
+		}
+		params.Properties.ARecords = aRecords
+
+	case armdns.RecordTypeAAAA:
+		aaaaRecords := make([]*armdns.AaaaRecord, len(record.Spec.Targets))
+		for i, target := range record.Spec.Targets {
+			aaaaRecords[i] = &armdns.AaaaRecord{IPv6Address: to.Ptr(target)}
 		}
+		params.Properties.AaaaRecords = aaaaRecords
+
 	case armdns.RecordTypeCNAME:
+		// CNAME can only have one target
```

Comment:
Take this as me thinking out loud, but now that we support multiple targets, I wonder if it wouldn't be preferable to return an error instead of just using the first target in the list. What I like about returning an error is that the user doesn't expect a record created as being successful.

A meaningful error message could tell the user that what they wanted to do couldn't be done and they could fix their error. Creating as it is now could possibly cause confusion down the road for the user (Why only 1 target is created here, I wanted 2...)?

What do you think?

Reply:
Happy to go with your thoughts on this one, sometimes it's much better to see logs and know what is going on, than scratching your head going "why is it only doing 1 entry?"

The reason I went with this as a fallback was your original code was selecting the first element, so wanted to mirror the same behaviours, 

Perhaps instead of erroring and not inserting a record, we select the first element but throw a warning in the logs?

Happy either way though! verbosity is always good

Reply:
Actually, the more I think about this the more I think throwing an error is best, that will show on the event history on the CRD and it should be very clear to the user what happened then

I'll update :)

Reply:
Should all be done :)